app/src/main/java/com/omkarmoghe/pokemap/views/MainActivity.java

```diff
@@ -170,6 +170,12 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
     }
 
     private void startNotificationService(){
+
+        // check for if the service is already running
+        if (PokemonNotificationService.isRunning()) {
+            stopNotificationService();
```

Comment:
They both are quick fixes. Just having it check to see if the service is running, if it is it does nothing is better than stopping it and starting it again. In fact you get better performance if you keep it running instead of stopping and starting it.